bigtable/google/cloud/bigtable/table.py

```diff
@@ -532,8 +544,48 @@ def _do_mutate_retryable_rows(self):
         return self.responses_statuses
 
 
+class _RetryableReadRows(object):
+    """A callable worker that can retry to read rows with transient errors.
+
+    This class is a callable that can retry reading rows that result in
+    transient errors.
+    """
+
+    def __init__(self, client, table_name, last_scanned_key,
+                 end_key, filter_, limit):
+        self.client = client
+        self.table_name = table_name
+        self.last_scanned_key = last_scanned_key
+        self.end_key = end_key
+        self.filter_ = filter_
+        self.limit = limit
+
+    def __call__(self):
+        from google.cloud.bigtable.row import _retry_commit_exception
+
+        read_rows = functools.partial(self._do_read_retryable_rows)
+        retry_ = Retry(
+            predicate=_retry_commit_exception,
+            deadline=30)
```

Comment:
A deadline doesn't make sense for a streaming read.  If we absolutely need one as a default, then perhaps something large like 20 minutes would do. This should be part of any ReadRetryOptions object we create.

Users can stream a lot of data, and might even read for an hour or so.

Reply:
For `unary-stream` calls, retry applies to **establishing** the stream. Basically, this deadline says "if I can't start a stream after trying for 30 seconds, give up" not "if the stream is open for 30 seconds, give up".

Reply:
The retry for read rows is under test on an endurance test server and we received the following StatusCode.DEADLINE_EXCEEDED exception.  It seems that Retry terminates the connection and causes our test code to abort, so there is more testing to be done.

It is hard to know what conditions cause this exception on the network, and whether this should be recoverable in Retry.  I suspect that the retry deadline of 30 seconds is too short.  We need to go into the Retry code itself to see how it treats this exception.

```
 File "/home/jenkins/.local/lib/python3.5/site-packages/google/cloud/bigtable/row_data.py", line 298, in read_rows
    response = six.next(self._response_iterator)
  File "/home/jenkins/.local/lib/python3.5/site-packages/grpc/_channel.py", line 348, in __next__
    return self._next()
  File "/home/jenkins/.local/lib/python3.5/site-packages/grpc/_channel.py", line 342, in _next
    raise self
grpc._channel._Rendezvous: <_Rendezvous of RPC that terminated with (StatusCode.DEADLINE_EXCEEDED, Error while reading table 'projects/grass-clump-479/instances/shared-perf/tables/scantest' : Response was not consumed in time; terminating connection. (Possible causes: slow client data read or network problems))>
```